app/src/test/kotlin/io/eugenethedev/taigamobile/repositories/TasksRepositoryTest.kt

```diff
@@ -0,0 +1,945 @@
+                            && !it.isClosed
+                }
+                .map {
+                    TestCommonTask(
+                        title = it.title,
+                        assignee = it.assignedTo,
+                        isClosed = it.isClosed
+                    )
+                }
+
+            val stories = project.userstories
+                .filter {
+                    (TestData.activeUser in it.watchers || TestData.activeUser in it.comments.map { it.author })
+                            && !it.isClosed
+                }
+                .map {
+                    TestCommonTask(
+                        title = it.title,
+                        assignee = it.assignedTo,
+                        isClosed = it.isClosed
+                    )
+                }
+
+            val tasks = getTestTasks(index)
+                .filter {
+                    (TestData.activeUser in it.watchers || TestData.activeUser in it.comments.map { it.author })
+                            && !it.isClosed
+                }
+                .map {
+                    TestCommonTask(
+                        title = it.title,
+                        assignee = it.assignedTo,
+                        isClosed = it.isClosed
+                    )
+                }
+
+            val issue = project.issues
+                .filter {
+                    (TestData.activeUser in it.watchers || TestData.activeUser in it.comments.map { it.author })
+                            && !it.isClosed
+                }
+                .map {
+                    TestCommonTask(
+                        title = it.title,
+                        assignee = it.assignedTo,
+                        isClosed = it.isClosed
+                    )
+                }
+            totalTestCommonTasks += epics + stories + tasks + issue
+        }
+        totalTestCommonTasks.sortBy { it.title }
+        listCommonTasks.forEachIndexed { index, data ->
+            assertEquals(
+                expected = totalTestCommonTasks[index].title,
+                actual = data.title
+            )
+            assertEquals(
+                expected = totalTestCommonTasks[index].assignee?.fullName,
+                actual = data.assignee?.fullName
+            )
+            assertEquals(
+                expected = totalTestCommonTasks[index].isClosed,
+                actual = data.isClosed
+            )
+        }
+    }
+
+    @Test
+    fun `test get epics`() = runBlocking {
+        val epics = tasksRepository.getEpics(1, FiltersData()).sortedBy { it.title }
+        val testEpics = TestData.projects[0].epics.sortedBy { it.title }
+        val epicsWithFilter = tasksRepository.getEpics(
+            page = 1,
+            filters = FiltersData(
+                createdBy = listOf(
+                    UsersFilter(
+                        id = 0,
```

Comment:
No, this is wrong way of using filters. If you want to test `getEpics` with filters too, you need to retrieve these filters from repository (again, hardcoded data, we cannot be sure)